### PR TITLE
remove set_standard_output_variables macro, replace with standard_project* functions

### PR DESCRIPTION
**Pull request by yeswalrus**
"There's a bunch of stuff that set_standard_output_variables did that was beyond the scope of the name, and a number of them actually produced incorrect behavior if they weren't set prior to the first call to project. To ensure they're called in the correct order, change it into a wrapper around project(). Also changes from a macro to a function to avoid namespace leakage, and enforce the setting of the version property."

Supercedes https://github.com/leapmotion/standard/pull/3

### CODE REVIEW

File: standard/StandardProject.cmake [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  directory convention popular on Gnu.
+* Ensures that ARM and 64-bit code are built as position-independent code
+* Verifies that the compiler actually supports C++11
+* Sets the correct flags to enable C++11 on all platforms
+* Ensures compilation happens with visibility set to hidden by default
+* Builds fat binaries on mac
+* Sets standard_BUILD* variables for consumption later down the line.
+* Enforces the project has a VERSION set.
+]]
+function(standard_project project_name)
```
Comment (codemercenary) [inline standard/StandardProject.cmake:16]: I know you need to do things before _and_ after the `project` command, but I prefer being able to see the `project` command explicitly in my project.  Can you break this into a `standard_before_project` and `standard_after_project` command pair instead?

Comment (yeswalrus) [inline standard/StandardProject.cmake:16]: Done, though I used preinit and postinit

File: standard/Standard.cmake [hunk 2]
```diff
@@ -2,12 +2,11 @@
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR})
 
 # Need these for the rest of these scripts to work properly
-include(CTest)
```
Comment (codemercenary) [inline standard/Standard.cmake:5]: ?

Comment (yeswalrus) [inline standard/Standard.cmake:5]: Standard must be included before project() so we can use standard_project, and CTest must be included after the first call to project(). To address this, CTest is now included by standard_project_postinit

**Pull request merged**